lib/spack/spack/concretize.py

```diff
@@ -302,7 +302,7 @@ def concretize_compiler(self, spec):
            link to this one, to maximize compatibility.
         """
         # Pass on concretizing the compiler if the target is not yet determined
```

Comment:
update this comment to say we want BOTH OS and target before picking a compiler.